src/main/java/org/apache/flink/benchmark/MultipleInputBenchmark.java

```diff
@@ -36,18 +50,20 @@
 import org.openjdk.jmh.runner.options.OptionsBuilder;
 import org.openjdk.jmh.runner.options.VerboseMode;
 
-public class MultipleInputBenchmark extends BenchmarkBase {
+import java.util.concurrent.CompletableFuture;
 
+public class MultipleInputBenchmark extends BenchmarkBase {
 	public static final int RECORDS_PER_INVOCATION = TwoInputBenchmark.RECORDS_PER_INVOCATION;
 	public static final int ONE_IDLE_RECORDS_PER_INVOCATION = TwoInputBenchmark.ONE_IDLE_RECORDS_PER_INVOCATION;
+	public static final int CHAINED_IDLE_RECORDS_PER_INVOCATION = 3000;
```

Comment:
Why only 3000? That's 5000 times less then the other benchmarks here, so something sound very wrong here.

Also are you sure https://github.com/apache/flink/pull/17303 is fixing the issue? I've tried it locally and couldn't see a difference in this benchmark with/without this change. What are the results that you have meassured?

Reply:
1. I forgot to move a flag from the UT and thus the chaining was not applied in the benchmark, sorry for that.
2. Yes, I am sure it fixes it. With the flag you can see a difference
3. I picked a significantly smaller number to give the benchmark a chance to finish with the old code. With 6 digit number of records, I am afraid it would not finish (with 5 digits it already goes into minutes for a single iteration)

Reply:
3000 records
* without the fix:
```
Result "org.apache.flink.benchmark.MultipleInputBenchmark.multiInputChainedIdleSource":
  4.352 ±(99.9%) 0.174 ops/ms [Average]
  (min, avg, max) = (4.003, 4.352, 5.241), stdev = 0.260
  CI (99.9%): [4.179, 4.526] (assumes normal distribution)
```
* with the fix
```
Result "org.apache.flink.benchmark.MultipleInputBenchmark.multiInputChainedIdleSource":
  7.802 ±(99.9%) 0.082 ops/ms [Average]
  (min, avg, max) = (7.608, 7.802, 8.243), stdev = 0.123
  CI (99.9%): [7.720, 7.884] (assumes normal distribution)
```

Reply:
Why is the performance so poor even after the fix? 7000 records/s is abysmal. That's still 2000-3000 times slower than the other benchmarks in this file.

Reply:
Please remember the test time includes job submission. Moreover the results are returned via an iterator over an RPC/HTTP from the jobmaster. The test is not comparable in any sense to all the others.

Reply:
If you have concerns towards the benchmark itself, I suggest merging the fix and create a separate followup for the benchmark.

Reply:
With throughput so low, you are not benchmarking any performance critical code path. I'm not sure what's causing this problems, maybe indeed collecting the results, but whatever it is, it's the performance bottleneck, which your benchmark is benchmarking. And if collecting results via HTTP request is that bottleneck, that's definitely not what we want to benchmark. 

On the other hand, something might be still very wrong with this code, and the performance regression might be still out there. Without properly working benchmark how do you know you have fixed the performance regression?